File: fuzzy-mobile-master/lib/common/model/assesment_model.dart
import 'dart:convert';

class Assesment {
  final int? nis;
  final String? fullname;
  int? significant;
  int? helpful;

  Assesment({
    this.nis,
    this.fullname,
    this.significant,
    this.helpful,
  });

  factory Assesment.fromJson(Map<String, dynamic> json) => Assesment(
        nis: json['nis'],
        fullname: json['fullname'],
        significant: json['significant'],
        helpful: json['helpful'],
      );

  factory Assesment.fromRawJson(String str) => Assesment.fromJson(json.decode(str));

  Assesment copyWith({
    int? nis,
    String? fullname,
    int? significant,
    int? helpful,
  }) =>
      Assesment(
        nis: nis ?? this.nis,
        fullname: fullname ?? this.fullname,
        significant: significant ?? this.significant,
        helpful: helpful ?? this.helpful,
      );

  Map<String, dynamic> toJson() => {
        'nis': nis,
        'fullname': fullname,
        'significant': significant,
        'helpful': helpful,
      };

  String toRawJson() => json.encode(toJson());
}

class AssesmentModel {
  final String? assesmentId;
  final String? groupName;
  final String? subjectName;
  final int? nis;
  final List<Assesment>? assesments;
  final DateTime? dateCreated;

  AssesmentModel({
    this.assesmentId,
    this.groupName,
    this.subjectName,
    this.nis,
    this.assesments,
    this.dateCreated,
  });

  factory AssesmentModel.fromJson(Map<String, dynamic> json) => AssesmentModel(
        assesmentId: json['assesment_id'],
        groupName: json['group_name'],
        subjectName: json['subject_name'],
        nis: json['nis'],
        assesments: json['assesments'] == null ? [] : List<Assesment>.from(json['assesments']!.map((x) => Assesment.fromJson(x))),
        dateCreated: json['date_created'] == null ? null : DateTime.parse(json['date_created']),
      );

  factory AssesmentModel.fromRawJson(String str) => AssesmentModel.fromJson(json.decode(str));

  AssesmentModel copyWith({
    String? assesmentId,
    String? groupName,
    String? subjectName,
    int? nis,
    List<Assesment>? assesments,
    DateTime? dateCreated,
  }) =>
      AssesmentModel(
        assesmentId: assesmentId ?? this.assesmentId,
        groupName: groupName ?? this.groupName,
        subjectName: subjectName ?? this.subjectName,
        nis: nis ?? this.nis,
        assesments: assesments ?? this.assesments,
        dateCreated: dateCreated ?? this.dateCreated,
      );

  Map<String, dynamic> toJson() => {
        'assesment_id': assesmentId,
        'group_name': groupName,
        'subject_name': subjectName,
        'nis': nis,
        'assesments': assesments == null ? [] : List<dynamic>.from(assesments!.map((x) => x.toJson())),
        'date_created': "${dateCreated!.year.toString().padLeft(4, '0')}-${dateCreated!.month.toString().padLeft(2, '0')}-${dateCreated!.day.toString().padLeft(2, '0')}",
      };

  String toRawJson() => json.encode(toJson());
}

File: fuzzy-go-master/pkg/common/model/assesment_model.go
package common_model

type Assesment struct {
	AssesmentID string `json:"assesment_id" gorm:"primaryKey" binding:"required" `
	GroupID     string `json:"group_id" gorm:"foreignkey:Group" binding:"required" `
	SubjectID   string `json:"subject_id" gorm:"foreignkey:Subject" binding:"required" `
	NIS         int    `json:"nis" gorm:"foreignkey:Student" binding:"required" `
	Assesments  string `json:"assesments"`
	DATECREATED string `json:"date_created" gorm:"column:date_created"`
}

type AssesmentInput struct {
	GroupID     string          `json:"group_id" gorm:"foreignkey:Group" binding:"required" `
	SubjectID   string          `json:"subject_id" gorm:"foreignkey:Subject" binding:"required" `
	NIS         int             `json:"nis" gorm:"foreignkey:Student" binding:"required" `
	Assesments  []PoinAssesment `json:"assesments" binding:"required"`
	DATECREATED string          `json:"date_created"`
}

type AssesmentShow struct {
	AssesmentID string          `json:"assesment_id" gorm:"primaryKey" binding:"required" `
	GroupName   string          `json:"group_name"`
	SubjectName string          `json:"subject_name"`
	NIS         int             `json:"nis" `
	Assesments  []PoinAssesment `json:"assesments"`
	DATECREATED string          `json:"date_created"`
}

type AssesmentUpdate struct {
	AssesmentID string          `json:"assesment_id" gorm:"primaryKey" binding:"required" `
	Assesments  []PoinAssesment `json:"assesments"`
}

type PoinAssesment struct {
	NIS         int    `json:"nis" gorm:"foreignkey:Student" binding:"required" `
	Fullname    string `json:"fullname"`
	Significant *int   `json:"significant"`
	Helpful     *int   `json:"helpful"`
}
